lib/apiservers/engine/backends/container.go

```diff
@@ -421,49 +421,46 @@ func (c *ContainerBackend) ContainerExecStart(ctx context.Context, eid string, s
 		actor := CreateContainerEventActorWithAttributes(vc, map[string]string{})
 		EventService().Log(event, eventtypes.ContainerEventType, actor)
 
-		// no need to attach for detached case
-		if !attach {
-			op.Debugf("Detached mode. Returning early.")
-			return nil
-		}
+		if attach {
+			EventService().Log(containerAttachEvent, eventtypes.ContainerEventType, actor)
+			ca := &backend.ContainerAttachConfig{
+				UseStdin:  ec.OpenStdin,
+				UseStdout: ec.OpenStdout,
+				UseStderr: ec.OpenStderr,
+			}
 
-		EventService().Log(containerAttachEvent, eventtypes.ContainerEventType, actor)
-		ca := &backend.ContainerAttachConfig{
-			UseStdin:  ec.OpenStdin,
-			UseStdout: ec.OpenStdout,
-			UseStderr: ec.OpenStderr,
-		}
+			// set UseStderr to false for Tty case as we merge stdout and stderr
+			if ec.Tty {
+				ca.UseStderr = false
+			}
 
-		// set UseStderr to false for Tty case as we merge stdout and stderr
-		if ec.Tty {
-			ca.UseStderr = false
-		}
+			ac := &proxy.AttachConfig{
+				ID: eid,
+				ContainerAttachConfig: ca,
+				UseTty:                ec.Tty,
+				CloseStdin:            true,
+			}
 
-		ac := &proxy.AttachConfig{
-			ID: eid,
-			ContainerAttachConfig: ca,
-			UseTty:                ec.Tty,
-			CloseStdin:            true,
-		}
+			err = c.streamProxy.AttachStreams(ctx, ac, stdin, stdout, stderr)
+			if err != nil {
+				if _, ok := err.(engerr.DetachError); ok {
+					op.Infof("Detach detected, tearing down connection")
 
-		err = c.streamProxy.AttachStreams(ctx, ac, stdin, stdout, stderr)
-		if err != nil {
-			if _, ok := err.(engerr.DetachError); ok {
-				op.Infof("Detach detected, tearing down connection")
+					// QUESTION: why are we returning DetachError? It doesn't seem like an error
```

Comment:
Follow up on this question - this really doesn't seem like an error, but an expected path.